CMakeLists.txt

```
@@ -13,9 +13,28 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(Eigen3 REQUIRED)
 
-install(DIRECTORY include/${PROJECT_NAME}/
-        DESTINATION include/${PROJECT_NAME}
-        FILES_MATCHING PATTERN "*.h")
+add_library(${PROJECT_NAME} INTERFACE
+  include/${PROJECT_NAME}/eigen_stl_containers.h
+  include/${PROJECT_NAME}/eigen_stl_map_container.h
+  include/${PROJECT_NAME}/eigen_stl_vector_container.h
+)
+
+ament_target_dependencies(${PROJECT_NAME} Eigen3)
```

I'd recommend using `target_link_libraries()` in new code. https://github.com/ament/ament_cmake/issues/292

---

I will fix that. Ty

---

Oh, I missed this one.  If you could change this to:

```suggestion
target_link_libraries(${PROJECT_NAME} INTERFACE Eigen3::Eigen3)
```

I think that should do it.